R/oauth-init.R

```diff
@@ -10,15 +10,16 @@
 #' @export
 #' @keywords internal
 init_oauth1.0 <- function(endpoint, app, permission = NULL,
-                          is_interactive = interactive()) {
+                          is_interactive = interactive(),
+                          private_key = NULL) {
 
-  oauth_sig <- function(url, method, token = NULL, token_secret = NULL, ...) {
-    oauth_header(oauth_signature(url, method, app, token, token_secret, ...,
+  oauth_sig <- function(url, method, token = NULL, token_secret = NULL, private_key = NULL, ...) {
+    oauth_header(oauth_signature(url, method, app, token, token_secret, private_key, ...,
       callback = oauth_callback()))
   }
 
   # 1. Get an unauthorized request token
-  response <- GET(endpoint$request, oauth_sig(endpoint$request, "GET"))
```

Comment:
Yeah, I'm hyper sensitive to not wanting to introduce regressions with this change, though the [RFC5849](http://tools.ietf.org/html/rfc5849#section-2.1) and [its predecessor](http://oauth.net/core/1.0a/#rfc.section.6.3.1) are fairly prescriptive that it should be `POST` unless otherwise noted:

```
   The client obtains a set of temporary credentials from the server by
   making an authenticated (Section 3) HTTP "POST" request to the
   Temporary Credential Request endpoint (unless the server advertises
   another HTTP request method for the client to use).
```

And:

```
To request an Access Token, the Consumer makes an HTTP request to the
Service Provider's Access Token URL. The Service Provider documentation
specifies the HTTP method for this request, and HTTP POST is RECOMMENDED.
The request MUST be signed per Signing Requests, and contains the
following parameters:
```